sdk/fetcher/fetcher.go

```diff
@@ -0,0 +1,32 @@
+package fetcher
+
+import (
+	"encoding/json"
+	"iter"
+)
+
+func YieldError(err error) iter.Seq2[map[string]any, error] {
+	return func(yield func(map[string]any, error) bool) {
+		if !(yield(nil, err)) {
+			return
+		}
+	}
+}
+
+func YieldMap[T any](marshaller func(any) ([]byte, error), objects []T) iter.Seq2[map[string]any, error] {
+	return func(yield func(map[string]any, error) bool) {
+		for _, object := range objects {
+			groupBytes, err := marshaller(object)
```

Comment:
This looks like a copy/paste error. This code doesn't deal with groups. It's generic.